src/routers/task.router.ts

```diff
@@ -46,8 +40,32 @@ taskRouter.get("/tasks/initialize", async (request, response) => {
 });
 
 taskRouter.get("/tasks", async (request, response) => {
-    const tasks = await TaskModel.find();
-    response.send(tasks);
+
+    const tasks = await TaskModel.find();    
+    response.send(tasks.map(task => toBasicTask(task)));
+
+});
+
+taskRouter.get("/tasks/:taskId", async (request, response) => {
+    const task = await TaskModel.findOne({ _id: request.params.taskId });
+    response.send(task);
+});
+
+taskRouter.post("/tasks", async (request, response) => {
+    // Validate that new task title is unique
+    const taskTitles = (await TaskModel.find()).map(task => task.title);
+    const requestBodyTaskTitle = request.body.title;
+    if (taskTitles.includes(requestBodyTaskTitle)) {
+        response.status(400).send({error: "Task title must be unique"});
+        return;
+    }
+    const task = await TaskModel.create(request.body);
+    response.send(task);
+});
+
+taskRouter.delete("/tasks/:taskId", async (request, response) => {
+    await TaskModel.deleteOne({ _id: request.params.taskId });
```

Comment:
you could send back the new list of tasks, to automatically update the list once you delete a task